"Pull Requests" and "Issues" have been showing "loading..." for hours

<!-- ⚠️⚠️ Do Not Delete This! bug_report_template ⚠️⚠️ -->
<!-- Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/ -->
<!-- 🕮 Read our guide about submitting issues: https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions -->
<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Insiders build to see if your issue has already been fixed: https://code.visualstudio.com/insiders/ -->
<!-- 💡 Instead of creating your report here, use 'Report Issue' from the 'Help' menu in VS Code to pre-fill useful information. -->
- VS Code Version: 1.54.2
- OS Version: 10.0.19042 Build 19042

Steps to Reproduce:

1. Install GitHub Pull Requests and Issues in VSCode
2. Authorize with Github
3. Go to the Github Section with the tab on the left side
4. Pull Requests and Issues have been showing "loading..." for hours

<!-- 🔧 Launch with `code --disable-extensions` to check. -->
Does this issue occur when all extensions are disabled?: No

<!-- 🪓 If you answered No above, use 'Help: Start Extension Bisect' from Command Palette to try to identify the cause. -->
<!-- 📣 Issues caused by an extension need to be reported direct to the extension publisher. The 'Help > Report Issue' dialog can assist with this. -->

I have the same issue and see the 2 warnings and 2 errors below in the Developer console:

workbench.desktop.main.js:62 [Extension Host] rejected promise not handled within 1 second: Error: Git model not found
v @ workbench.desktop.main.js:62

workbench.desktop.main.js:62 [Extension Host] stack trace: Error: Git model not found	at a.getAPI (c:\Users\User\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\git\dist\main.js:54:45368)	at new Da (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1008:18628)	at Function.createProvider (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1008:19195)	at Dp (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1008:25145)	at Eg (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1020:23211)	at processTicksAndRejections (internal/process/task_queues.js:97:5)
v @ workbench.desktop.main.js:62

workbench.desktop.main.js:2084 [[object Object]]Git model not found
$onExtensionRuntimeError @ workbench.desktop.main.js:2084

workbench.desktop.main.js:2084 Error: Git model not found
	at a.getAPI (c:\Users\User\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\git\dist\main.js:54:45368)
	at new Da (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1008:18628)
	at Function.createProvider (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1008:19195)
	at Dp (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1008:25145)
	at Eg (c:\Users\User\.vscode\extensions\github.vscode-pull-request-github-0.24.0\dist\extension.js:1020:23211)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
$onExtensionRuntimeError @ workbench.desktop.main.js:2084

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -234,7 +234,7 @@ async function deferredActivate(context: vscode.ExtensionContext, apiImpl: GitAp
 		await credentialStore.initialize(AuthProvider['github-enterprise']);
 	}
 
-	const builtInGitProvider = registerBuiltinGitProvider(credentialStore, apiImpl);
+	const builtInGitProvider = await registerBuiltinGitProvider(credentialStore, apiImpl);
 	if (builtInGitProvider) {
 		context.subscriptions.push(builtInGitProvider);
 	}

--- a/src/gitProviders/api.ts
+++ b/src/gitProviders/api.ts
@@ -14,11 +14,11 @@ export function registerLiveShareGitProvider(apiImpl: API): LiveShareManager {
 	return liveShareManager;
 }
 
-export function registerBuiltinGitProvider(
+export async function registerBuiltinGitProvider(
 	_credentialStore: CredentialStore,
 	apiImpl: API,
-): vscode.Disposable | undefined {
-	const builtInGitProvider = BuiltinGitProvider.createProvider();
+): Promise<vscode.Disposable | undefined> {
+	const builtInGitProvider = await BuiltinGitProvider.createProvider();
 	if (builtInGitProvider) {
 		apiImpl.registerGitProvider(builtInGitProvider);
 		return builtInGitProvider;

--- a/src/gitProviders/builtinGit.ts
+++ b/src/gitProviders/builtinGit.ts
@@ -30,7 +30,15 @@ export class BuiltinGitProvider implements IGit, vscode.Disposable {
 
 	private constructor(extension: vscode.Extension<GitExtension>) {
 		const gitExtension = extension.exports;
-		this._gitAPI = gitExtension.getAPI(1);
+
+		try {
+			this._gitAPI = gitExtension.getAPI(1);
+		} catch (e) {
+			// The git extension will throw if a git model cannot be found, i.e. if git is not installed.
+			vscode.window.showErrorMessage('Activating the Pull Requests and Issues extension failed. Please make sure you have git installed.');
+			throw e;
+		}
+
 		this._disposables = [];
 		this._disposables.push(this._gitAPI.onDidCloseRepository(e => this._onDidCloseRepository.fire(e as any)));
 		this._disposables.push(this._gitAPI.onDidOpenRepository(e => this._onDidOpenRepository.fire(e as any)));
@@ -38,9 +46,10 @@ export class BuiltinGitProvider implements IGit, vscode.Disposable {
 		this._disposables.push(this._gitAPI.onDidPublish(e => this._onDidPublish.fire(e)));
 	}
 
-	static createProvider(): BuiltinGitProvider | undefined {
+	static async createProvider(): Promise<BuiltinGitProvider | undefined> {
 		const extension = vscode.extensions.getExtension<GitExtension>('vscode.git');
 		if (extension) {
+			await extension.activate();
 			return new BuiltinGitProvider(extension);
 		}
 		return undefined;

